device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/Application.java

```diff
@@ -0,0 +1,168 @@
+    private static final int SLEEP_DURATION_IN_SECS = 60;
+    private static AtomicBoolean stopRunning = new AtomicBoolean(false);
+
+    static boolean reconnectIndefinitely = true;
+    static List<String> failedMessageListOnClose = new ArrayList<>(); // List of messages that failed on close
+    private static CountDownLatch countDownLatch = new CountDownLatch(2);
+    private static DeviceClientManager deviceClientManager;
+
+    /**
+     * Sends a number of messages to an IoT. Default protocol is to use AMQP transport.
+     *
+     */
+    public static void main(String[] args)
+            throws IOException, URISyntaxException, InterruptedException {
+        log.debug("Starting the sample...");
+        log.debug("Using communication protocol: {}", protocol.name());
+
+        DeviceClient client = new DeviceClient(deviceConnectionString, protocol);
+        log.debug("Successfully created an IoT Hub client.");
+
+        deviceClientManager = new DeviceClientManager(client);
+        deviceClientManager.setOperationTimeout(DEVICE_OPERATION_TIMEOUT_IN_MINUTES);
+
+        deviceClientManager.connect();
+        log.debug("Opened connection to IoT Hub.");
+
+        log.debug("Setting C2D message handler...");
+        deviceClientManager.setMessageCallback(new MessageCallback(), new Object());
+
+        log.debug("Setting method handler...");
+        deviceClientManager.subscribeToDeviceMethod(new SampleDeviceMethodCallback(), null, new DeviceMethodStatusCallBack(), null);
+
+        log.debug("Start device Twin and get remaining properties...");
+        // Properties already set in the Service will shows up in the generic onProperty callback, with value and version.
+        Succeed.set(false);
+        client.startDeviceTwin(new DeviceTwinStatusCallBack(), null, new onProperty(), null);
+        do {
+            Thread.sleep(1000);
+        }
+        while (! Succeed.get());
+
+        log.debug("Subscribe to Desired properties on device Twin: PropertyKey={}...", TwinPropertyKey);
+        Map<Property, Pair<TwinPropertyCallBack, Object>> propertyPairMap =
+                singletonMap(new Property(TwinPropertyKey, null), new Pair<TwinPropertyCallBack, Object>(new onProperty(), null));
+        deviceClientManager.subscribeToTwinDesiredProperties(propertyPairMap);
+
+        log.debug("Start sending telemetry and reported property updates...");
+        new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                updateReportedProperties();
+                countDownLatch.countDown();
+            }
+        }).start();
+
+        new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                sendTelemetry();
+                countDownLatch.countDown();
+            }
+        }).start();
+
+        countDownLatch.await(MAX_APPLICATION_RUN_DURATION_IN_MINUTES, TimeUnit.MINUTES);
```

Comment:
return the value of the countdown latch to indicate that the latch was released vs time out

Reply:
There is no difference from the sample's point of view if the latch was released or if it timed out. So returning the value of await is not very useful here.

Reply:
Also, I am removing the `MAX_APPLICATION_RUN_DURATION_IN_MINUTES`. It makes sense to have a single condition for the loop `NUM_REQUESTS`.